openquake/server/views.py

```diff
@@ -368,10 +381,16 @@ def calc_results(request, calc_id):
         * type (hazard_curve, hazard_map, etc.)
         * url (the exact url where the full result can be accessed)
     """
+    user_name = "platform"
+    if hasattr(request, 'user'):
+        if request.user.is_authenticated():
+            user_name = request.user.username
+
     # If the specified calculation doesn't exist OR is not yet complete,
     # throw back a 404.
     try:
-        oqjob = oqe_models.OqJob.objects.get(id=calc_id)
+        oqjob = oqe_models.OqJob.objects.get(id=calc_id,
+                                             user_name=user_name)
```

Comment:
What happens if the user is looking at a calculation of another user? `oqe_models.OqJob.objects.get` gives an ugly DoesNotExist error. We should probably do

```
oqjob = oqe_models.OqJob.objects.get(id=calc_id)
if oqjob.username != username:
      return ...
```